### ISSUE TITLE
h3ToParent() issue on latest stable release 22.2.2.1

### ISSUE THREAD

**Issue by ray-at-github** [ClickHouse/ClickHouse#34708]
Latest stable release 22.2.2.1

This query works fine:

```sql
select h3ToParent(641573946153969375, 1);
```

This doesn't. It used to work on 22.1.3.7
```sql
select h3ToParent(641573946153969375, arrayJoin([1,2]));
```

Error message:
```
Code: 44. DB::Exception: Received from localhost:9000. DB::Exception: Illegal type UInt64 of argument 1 of function h3ToParent. Must be UInt64.. (ILLEGAL_COLUMN)
```

**Comment by Tiaonmmn** [comment]
Not only h3ToParent,but also geoToH3(toFloat64(xxx),toFloat64(yyy),8).Errors occured:Illegal type UInt8 of argument 3 of function geoToH3. Must be UInt8.

**Comment by bharatnc** [comment]
> @bharatnc Please add some tests for these functions. We will start reverting these pull requests one by one: https://github.com/ClickHouse/ClickHouse/pulls?q=is%3Apr+author%3Abharatnc+is%3Aclosed until the issue will be fixed.

I have a PR that fixes errors for queries involving const columns.

`geoToH3` which was failing benchmark was fixed in #34731 by @kitaisreal.

**Comment by ray-at-github** [comment]
This is off topic.. but @bharatnc I can see you have been completing the H3 function set progressively. Great work and I appreciate it! Just wondering if we plan to implement the polyfill() function, as this is the most anticipated function for me.

**Comment by bharatnc** [comment]
> This is off topic.. but @bharatnc I can see you have been completing the H3 function set progressively. Great work and I appreciate it! Just wondering if we plan to implement the polyfill() function, as this is the most anticipated function for me.

Yes, I have been adding functions by type (inspection, indexing etc) and in no specific order. I guess I can prioritize polyfill function next.

**Comment by ray-at-github** [comment]
> > This is off topic.. but @bharatnc I can see you have been completing the H3 function set progressively. Great work and I appreciate it! Just wondering if we plan to implement the polyfill() function, as this is the most anticipated function for me.
> 
> Yes, I have been adding functions by type (inspection, indexing etc) and in no specific order. I guess I can prioritize polyfill function next.

Great! Appreciate it.

**Comment by dynnikovda** [comment]
Hello! I have the same problem with geoToH3() function

If I do this query:
```sql
select geoToH3(longitude, latitude, toUInt8(8)) AS h3Index
from db_name
```
I get error message "_llegal type UInt8 of argument 3 of function geoToH3. Must be UInt8_"

But this one works fine!!!
```sql
select geoToH3(longitude, latitude, toUInt8(latitude/latitude*8)) AS h3Index
from db_name
```

**Comment by kitaisreal** [comment]
@dynnikovda could not reproduce. Please provide full example.
```sql
CREATE TABLE data_table (id UInt64, longitude Float64, latitude Float64) ENGINE=MergeTree ORDER BY id;
INSERT INTO data_table SELECT number, number, number FROM numbers(10);
SELECT geoToH3(longitude,  latitude, toUInt8(8)) AS h3Index FROM data_table
```
Results:
```
┌────────────h3Index─┐
│ 614553222213795839 │
│ 614552348391374847 │
│ 614555412668088319 │
│ 614554538768072703 │
│ 614790495813500927 │
│ 614048195802038271 │
│ 614047153853038591 │
│ 614047082918969343 │
│ 614054419345965055 │
│ 614054260742553599 │
└────────────────────┘
```

**Comment by dynnikovda** [comment]
@kitaisreal I launch your example
```sql
CREATE TABLE data_table (id UInt64, longitude Float64, latitude Float64) ENGINE=MergeTree ORDER BY id;
INSERT INTO data_table SELECT number, number, number FROM numbers(10);
SELECT geoToH3(longitude,  latitude, toUInt8(8)) AS h3Index FROM data_table
```
And I get error message:
```
Code: 44. DB::Exception: Illegal type UInt8 of argument 3 of function geoToH3. Must be UInt8. (ILLEGAL_COLUMN) (version 22.2.2.1)
```
But this one works fine:
```sql
CREATE TABLE data_table (id UInt64, longitude Float64, latitude Float64) ENGINE=MergeTree ORDER BY id;
INSERT INTO data_table SELECT number, number, number FROM numbers(10);
SELECT geoToH3(longitude,  latitude, toUInt8(longitude-longitude+8)) AS h3Index FROM data_table
```
Results:
```
┌────────────h3Index─┐
│ 614553222213795839 │
│ 614552348391374847 │
│ 614555412668088319 │
│ 614554538768072703 │
│ 614790495813500927 │
│ 614048195802038271 │
│ 614047153853038591 │
│ 614047082918969343 │
│ 614054419345965055 │
│ 614054260742553599 │
└────────────────────┘
```

**Comment by dynnikovda** [comment]
@kitaisreal  is it enough?

| name | value |
| :--- | :--- |
| VERSION\_FULL | ClickHouse 22.2.2.1 |
| VERSION\_DESCRIBE | v22.2.2.1-prestable |

**Comment by bharatnc** [comment]
Was this fix was ever backported? For example I had a similar fix for few S2 functions (#34745) and I could see that it was backported.

**Comment by alexey-milovidov** [comment]
I have checked and can confirm that all queries work perfectly.
Install the latest ClickHouse release as follows:
```
curl https://clickhouse.com/ | sh
```
